python/lsst/pipe/base/pipeline.py

```diff
@@ -58,6 +58,27 @@
 # ------------------------
 
 
+@dataclass
+class LabelSpecifier:
+    """A structure to specify a subset of labels to load
+
+    This structure may contain a set of lables to be used in subsetting a
+    pipeline, or a beginning and end point. Beggining or end may be empty in
+    which case the range will be a half open interval. Unlike python
+    iteration end bounds are *INCLUDED*. Note range based selection is not
+    well defined for pipelines that are not linear in nature, and correct
+    behavior is not guaranteed, or may vary from run to run.
```

Comment:
I want to deffer that to another ticket. Honestly I am not sure. I think it would involve changing toExpandedPipeline return a different kind of datastructure and change how it decides to order something, keeping this in mind.

Reply:
I am also thinking about how a user would know what the order is without some other knowledge of the pipeline. Just printing out the pipeline does not tell you the order. The `toExpandedPipeline` will return a partially (maybe fully) ordered iterator of `TaskDefs`, but they would need to be interactive for that. If they had the quantumGraph they could get the topoligically sorted graph of `TaskDefs`, and so maybe it is worth switching `toExpandedPipeline` to use that algorithm, and use that to order the pipeline as it is loaded, making toExpandedPipeline effectively just a `freeze` step. However `Pipeline`s can have configs applied, tasks added/removed etc all the way up to calling toExpandedPipeline which means we would have to call the reordering step each time the `Pipeline` is modified.

All these things are doable, but I think should go on a ticket on their own where it gets the time to think it though. 95% of the behavior we want is here in this ticket, and I think the interface we want is in this ticket (which is what is important for next week) so I would like to re-visit when there is time to do it appropriately.